### ISSUE TITLE
OSM editing plugin: "Node cannot be found or amenity is not a single node" error when Modifing POI

### ISSUE THREAD

**Issue by andrasfuchs** [osmandapp/Osmand#3480]
I'm using OsmAnd+ 2.5.4 with its OSM editing plugin.

Please take a look at this entry: 
![Screenshot of the map segment where this Starbucks is](https://cloud.githubusercontent.com/assets/910321/22615000/0c23bebc-eabe-11e6-90bd-3e04aeef3747.png)

Map: http://osm.org/go/4HtJD42SD?m=

This Starbucks is in its own building, on the island of Ko Samui, Thailand. It has the following tags:
name: Starbucks Coffee
amenity: cafe
building: yes
shop: coffee

I was unable to add the opening hours to it, because when I tapped on it, selected any of the 3 entries (building, coffee shop or cafe), tapped "modify POI", I got the "Node cannot be found or amenity is not a single node" error.

**Comment by sonora** [comment]
It is a known limitation that our plugin currently only handles node POIs, not area POIs.

**Comment by andrasfuchs** [comment]
Could you change the error message or disable this option then?
Are you planning to implement the area POI modification in the future?

**Comment by vshcherb** [comment]
Yes it is planned to let users modify tags on all possible objects.

**Comment by sonora** [comment]
Error text slightly amended for the interim (https://github.com/osmandapp/Osmand/commit/b98fbffd55e8c5c6ab687c8bb7c3c8b9fe945178).

**Comment by clembeb** [comment]
Hi,
Same needs as @andrasfuchs. The OSM plugin can only add/edit node POI it would be very nice to add/edit areas POI.

Thanks

**Comment by skylerbunny** [comment]
Ping on this: any chance this could be implemented?

I wouldn't be expecting all that much from a first pass: just to be able to modify tags of an area if I select one - nothing like 'modifying its area' or anything.

--

(Presumably pressing a spot on the map would test if a node/area/way is there, and:

1. If a node is detected, give the user the option to modify tags on the node. (a behavior done today IIRC).

2. _New:_ If a way or area but NOT a node is found in the spot, highlight the presumed object to edit, then give the user a choice: 'edit the POI tags of the way/area', (or) 'create a new POI node in that location'.

3. If neither is found, give the option to create a new POI in that spot (also a behavior done today.)

**Comment by clembeb** [comment]
This issue could be merged with this other issue "Allow edit tags for non-point objects  #5595"
implemented in the future milestone.
